pkg/build/builder/common.go

```diff
@@ -66,6 +66,25 @@ func buildInfo(build *api.Build) []KeyValue {
 	return kv
 }
 
+// updateBuildStatus updates the current status of the build only if there was
+// an error
+func updateBuildStatus(c client.BuildInterface, build *api.Build) {
```

Comment:
well, updateBuildRevision doesn't do that. it just logs the error message if there is one. so i sort of followed that pattern 
https://github.com/openshift/origin/blob/master/pkg/build/builder/common.go#L69